### PR TITLE
[MAINTENANCE] Correct typo in Checkpoint test method fixture

### PR DESCRIPTION
**Pull request by alexsherstinsky**
### Scope
* Current version had typographical errors in `Checkpoint` test fixture method; it is being corrected.
* We introduce two additional `great_expectations.yml` test fixture files, each containing the `fluent_datasource` section.

### Remarks
* Ideally, this pull request should be reviewed after #7670 has been merged.

Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], or [CONTRIB]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

Changes proposed in this pull request:
- JIRA: DX-469/DX-470
-
-

After submitting your PR, CI checks will run and @cla-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in GitHub issues or Slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -1125,7 +1125,7 @@ def titanic_v013_multi_datasource_multi_execution_engine_data_context_with_check
 
 
 @pytest.fixture
-def deterministic_asset_dataconnector_context(
+def deterministic_asset_data_connector_context(
```
Comment (Shinnnyshinshin) [inline tests/conftest.py:1130]: Just confirming : `data_connector` is the correct spelling and `dataconnector` is not?

File: tests/test_fixtures/great_expectations_no_block_no_fluent_datasources_stats_enabled.yml [hunk 2]
```diff
@@ -0,0 +1,56 @@
+# This is a basic configuration for testing.
+# It has comments that should be preserved.
+config_version: 3
+datasources:
```
Comment (Shinnnyshinshin) [inline tests/test_fixtures/great_expectations_no_block_no_fluent_datasources_stats_enabled.yml]: Are these fixtures used at the moment?

File: tests/conftest.py (hunk 1 above)
Comment (alexsherstinsky) [reply tests/conftest.py:1130]: @Shinnnyshinshin Yes -- indeed; hence the proposed change.  Thank you!

File: tests/test_fixtures/great_expectations_no_block_no_fluent_datasources.yml [hunk 3]
```diff
@@ -0,0 +1,44 @@
+# This is a basic configuration for testing.
+# It has comments that should be preserved.
+config_version: 3
```
Comment (Shinnnyshinshin) [inline tests/test_fixtures/great_expectations_no_block_no_fluent_datasources.yml]: Same question

File: tests/test_fixtures/great_expectations_no_block_no_fluent_datasources_stats_enabled.yml (hunk 2 above)
Comment (alexsherstinsky) [reply tests/test_fixtures/great_expectations_no_block_no_fluent_datasources_stats_enabled.yml]: @Shinnnyshinshin Not yet -- preparation for next PR.

File: tests/test_fixtures/great_expectations_no_block_no_fluent_datasources.yml (hunk 3 above)
Comment (alexsherstinsky) [reply tests/test_fixtures/great_expectations_no_block_no_fluent_datasources.yml]: @Shinnnyshinshin I am preparing them for the next addition.  Thanks.

File: tests/test_fixtures/great_expectations_no_block_no_fluent_datasources_stats_enabled.yml (hunk 2 above)
Comment (Shinnnyshinshin) [reply tests/test_fixtures/great_expectations_no_block_no_fluent_datasources_stats_enabled.yml]: If not, just curious why we are adding at this now

Comment (alexsherstinsky) [reply tests/test_fixtures/great_expectations_no_block_no_fluent_datasources_stats_enabled.yml]: They will be used in the next pull request.

File: tests/conftest.py [hunk 4]
```diff
@@ -1142,7 +1142,10 @@ def deterministic_asset_dataconnector_context(
     shutil.copy(
         file_relative_path(
             __file__,
-            "./test_fixtures/great_expectations_v013_no_datasource_stats_enabled.yml",
+            os.path.join(  # noqa: PTH118
```
Comment (Kilo59) [inline tests/conftest.py]: Please don't use os.path for this.
We have these linting rules for a reason.

Comment (alexsherstinsky) [reply tests/conftest.py]: @Kilo59 It is an improvement of what was there before.  I can modernize it of course.

Comment (alexsherstinsky) [reply tests/conftest.py]: I modernized it in a couple of places.

**Review by donaldheppner — APPROVED** [review]

**Pull request merged**